### PR TITLE
UTF-7 updates for .NET 5

### PR DESCRIPTION
**Pull request by gewarren**
- Contributes to dotnet/docs#19274.
- "The .NET Framework" -> ".NET Framework"
- Cleans up excess whitespace.

*Please ignore whitespace changes in the diff.*

[Example preview link (see Remarks section)](https://review.docs.microsoft.com/en-us/dotnet/api/system.text.encoding.getencoding?view=netcore-3.1&branch=pr-en-us-4586#System_Text_Encoding_GetEncoding_System_Int32_).

### CODE REVIEW

File: xml/System.Text/Encoding.xml [hunk 1]
```diff
@@ -3565,28 +3529,27 @@ On .NET Core, the <xref:System.Text.Encoding.Default%2A> property always returns
...
 
 - On .NET Core, if no encoding provider has been registered or if no registered encoding provider handles a `codepage` value of 0, it returns the <xref:System.Text.UTF8Encoding>.
 
 > [!NOTE]
->  Some unsupported code pages cause an <xref:System.ArgumentException> to be thrown, whereas others cause a <xref:System.NotSupportedException>. Therefore, your code must catch all exceptions indicated in the Exceptions section.
+> - Some unsupported code pages cause an <xref:System.ArgumentException> to be thrown, whereas others cause a <xref:System.NotSupportedException>. Therefore, your code must catch all exceptions indicated in the Exceptions section.
+> - In .NET 5.0 and later versions, the code page identifier `65000`, which represents UTF-7, is not supported.
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml:3546]: Newly added note.

File: xml/System.Text/Encoding.xml [hunk 2]
```diff
@@ -3654,22 +3617,23 @@ You can also supply a value of 0 for the `codepage` argument. Its precise behavi
...
+ In addition to the encodings that are natively available on .NET Core or that are intrinsically supported on a specific platform version of .NET Framework, the <xref:System.Text.Encoding.GetEncoding%2A> method returns any additional encodings that are made available by registering an <xref:System.Text.EncodingProvider> object. If the same encoding has been registered by multiple <xref:System.Text.EncodingProvider> objects, this method returns the last one registered.
 
-> [!NOTE]
->  The ANSI code pages can be different on different computers, or can be changed for a single computer, leading to data corruption. For the most consistent results, you should use Unicode, such as UTF-8 (code page 65001) or UTF-16, instead of a specific code page.
+In .NET 5.0 and later versions, the code page name `utf-7` is not supported.
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml:3633]: Newly added note.

File: xml/System.Text/Encoding.xml [hunk 3]
```diff
@@ -3754,29 +3718,28 @@ In the .NET Framework, the <xref:System.Text.Encoding.GetEncoding%2A> method rel
 ## Remarks
 
 > [!NOTE]
->  Some unsupported code pages cause the exception <xref:System.ArgumentException> to be thrown, whereas others cause <xref:System.NotSupportedException>. Therefore, your code must catch all exceptions indicated in the Exceptions section.
+> - Some unsupported code pages cause the exception <xref:System.ArgumentException> to be thrown, whereas others cause <xref:System.NotSupportedException>. Therefore, your code must catch all exceptions indicated in the Exceptions section. 
+> - In .NET 5.0 and later versions, the code page identifier `65000`, which represents UTF-7, is not supported.
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml:3722]: Newly added note.

File: xml/System.Text/Encoding.xml [hunk 4]
```diff
@@ -3854,17 +3817,17 @@ You can also supply a value of 0 for the `codepage` argument. Its precise behavi
...
 
- In addition to the encodings that are natively available on .NET Core or that are intrinsically supported on a specific platform version of the .NET Framework, the <xref:System.Text.Encoding.GetEncoding%2A> method returns any additional encodings that are made available by registering an <xref:System.Text.EncodingProvider> object. If the same encoding has been registered by multiple <xref:System.Text.EncodingProvider> objects, this method returns the last one registered.
+ In addition to the encodings that are natively available on .NET Core or that are intrinsically supported on a specific platform version of .NET Framework, the <xref:System.Text.Encoding.GetEncoding%2A> method returns any additional encodings that are made available by registering an <xref:System.Text.EncodingProvider> object. If the same encoding has been registered by multiple <xref:System.Text.EncodingProvider> objects, this method returns the last one registered.
+
+In .NET 5.0 and later versions, the code page name `utf-7` is not supported.
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml:3824]: Newly added note.

File: xml/System.Text/Encoding.xml [hunk 5]
```diff
@@ -68,173 +68,176 @@
...
+|57003|x-iscii-be|ISCII Bengali|✓|||
+|57004|x-iscii-ta|ISCII Tamil|✓|||
+|57005|x-iscii-te|ISCII Telugu|✓|||
+|57006|x-iscii-as|ISCII Assamese|✓|||
+|57007|x-iscii-or|ISCII Oriya|✓|||
+|57008|x-iscii-ka|ISCII Kannada|✓|||
+|57009|x-iscii-ma|ISCII Malayalam|✓|||
+|57010|x-iscii-gu|ISCII Gujarati|✓|||
+|57011|x-iscii-pa|ISCII Punjabi|✓|||
+|65000|utf-7|Unicode (UTF-7)|✓|✓||
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml:239]: Added new column for .NET 5 and left it unchecked.

Comment (GrabYourPitchforks) [reply xml/System.Text/Encoding.xml:239]: I don't see the table reflected at https://review.docs.microsoft.com/en-us/dotnet/api/system.text.encoding?branch=master&view=net-5.0, but the XML looks fine.

Comment (gewarren) [reply xml/System.Text/Encoding.xml:239]: It looks like the branch reverted to master in your URL. It's here: https://review.docs.microsoft.com/en-us/dotnet/api/system.text.encoding?branch=pr-en-us-4586&view=net-5.0#list-of-encodings.

**Pull request merged**